rustls/src/client/tls13.rs

```diff
@@ -162,6 +165,30 @@ pub(super) fn handle_server_hello(
     // the two halves will have different record layer protections.  Disallow this.
     cx.common.check_aligned_handshake()?;
 
+    let ech_status = match ech_context {
+        // ECH wasn't offered.
+        None => ech::Status::NotOffered,
+        Some(ech_context) => {
+            match ech_context.confirm_acceptance(
+                &mut key_schedule,
+                server_hello,
+                suite.common.hash_provider,
+            )? {
+                // The server rejected our ECH offer.
+                None => ech::Status::Rejected,
```

Comment:
Style nit: I'd move this boring case down.

Reply:
Done.